R/zchunk_LA161.Cstorage_china.R

```diff
@@ -0,0 +1,125 @@
+    return(c("L161.Cstorage_province"))
+  } else if(command == driver.MAKE) {
+
+    # Silence package checks
+    province <- grid_region <- Cost_2005USDtCO2 <- CO2_Mt <- MtC <- Cumul_MtC <- quantile <-
+      grade <- Cost_1990USDtC <- NULL
+
+    all_data <- list(...)[[1]]
+
+    # -----------------------------------------------------------------------------
+    # 1.Load required inputs
+    province_names_mappings <- get_data(all_data, "gcam-china/province_names_mappings")
+    Dooley_CCS_China        <- get_data(all_data, "gcam-china/CCS_provincial_data")
+    CCS_China_add           <- get_data(all_data, "gcam-china/CCS_provincial_data_add")
+    Unallocated_CStorage    <- get_data(all_data, "gcam-china/unallocated_CStorage_province")
+
+    # -----------------------------------------------------------------------------
+    # 2.perform computations
+    L161.Csupply_province <- Dooley_CCS_China %>%
+      # Add grid.region
+      left_join_error_no_match(province_names_mappings, by = "province.name") %>%
+      # Convert prices to 1990 USD per ton C and amounts to C
+      mutate(Cost_1990USDtC = Cost_2005USDtCO2 * emissions.CONV_C_CO2 / gdp_deflator(2005, 1990),
+             MtC = CO2_Mt * (1 / emissions.CONV_C_CO2)) %>%
+      arrange(province.name, Cost_2005USDtCO2) %>%
+    # Removing data for provinces with not enough onshore storage
+      filter(!(province.name %in% c("Fujian", "Guangdong", "Guangxi", "Qinghai")))
+
+    province_data <- L161.Csupply_province %>%
+      group_by(province.name) %>%
+      # Calculate cumulative sum, then filter to the quantiles
+      mutate(Cumul_MtC = cumsum(MtC))
+
+    L161.Cstorage_province <- province_data %>%
+      # use "type=1" to indicate that we want to return only exact matches to the data in the table.
+      # only take the (0, 0.2, 0.4, 0.6, 0.8, 1) points in the quartiles
+      filter(Cumul_MtC %in% quantile(Cumul_MtC, c(0, 0.2, 0.4, 0.6, 0.8, 1), type = 1)) %>%
+      group_by(province.name) %>%
+      mutate(# this will create grade equal to grade 1 for 0, grade 2 for 0.2, etc.
+             grade = paste("grade",1:6, sep = " "),
+             # from the cumulative totals, return to the marginal quantities associated with each grade
+             MtC = Cumul_MtC- lag(Cumul_MtC,1,0)) %>%
+      ungroup %>%
+      # Return from the list to a data frame with all necessary data
+      select(province.name, grade, Cost_1990USDtC, MtC) %>%
+      # Add cost grades to provinces without data
+      bind_rows(CCS_China_add) %>%
+      mutate(# Setting a minimum cost of 0 on CO2 storage and transport projects
+             Cost_1990USDtC = pmax(Cost_1990USDtC, 0))
+
+    # Manually changing cost curves to avoid grades 2 or 3 being 0. Value determined by linear interpolation. If the data changes in the future, will need to change this
+    L161.Cstorage_province$Cost_1990USDtC[which(L161.Cstorage_province$province.name== "Jilin" & L161.Cstorage_province$grade== "grade 2")]    <- 4.625
```

Comment:
continue with the pipe above, by using `mutate` + `replace`
`%>% mutate(Cost_1990USDtC = replace(Cost_1990USDtC, province.name == "Jilin" & grade == "grade 2", 4.625) %>%`

Reply:
get